### PR TITLE
Provide Tenant Invitations

### PR DESCRIPTION
**Pull request by majst01**
Same Idea as for Projects.

TODO:

Define how to create a new Tenant, basically a logged in Tenant should be able to create a new Tenant with a generated UUID and a choosen Name, and the creator will be the Tenant owner. After that he/she can invite others with this call.

There is already a TenanServiceCreateRequest, but for other purposes.

For the api-server implementation https://github.com/metal-stack/masterdata-api/pull/89 is required.

### CODE REVIEW

**Comment by Gerrit91** [comment]
With generated UUIDs for the tenant we need to change the token structure where we identify project permissions by trying to parse the subject as a UUID.

File: proto/api/v1/tenant.proto [hunk 1]
```diff
@@ -182,3 +248,80 @@ message TenantServiceDeleteResponse {
...
+// TenantServiceInviteGetResponse is the response payload to a get invite request
+message TenantServiceInviteGetResponse {
+  // Invite is the invite
+  TenantInvite invite = 1;
+}
+
+// TenantServiceRemoveMemberRequest is used to remove a member from a tenant
+message TenantServiceRemoveMemberRequest {
+  // Tenant is the uuid of the tenant
+  string tenant = 1 [(validate.rules).string.uuid = true];
```
Comment (Gerrit91) [inline proto/api/v1/tenant.proto:296]: If we want to maintain old tenant ID naming convention, we cannot validate UUIDs here.

Comment (majst01) [reply proto/api/v1/tenant.proto]: removed this validation

File: proto/api/v1/tenant.proto [hunk 2]
```diff
@@ -186,6 +195,19 @@ message TermsAndConditionsUpdate {
   optional google.protobuf.Timestamp when = 2;
 }
 
+// TenantServiceListRequest is the request payload of the tenant list request
+message TenantServiceListRequest {
+  // Login of the tenant
+  string login = 1;
+  // Id filters tenants by id
+  string id = 2;
```
Comment (majst01) [inline proto/api/v1/tenant.proto:203]: All should be optional

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**